src/plugins/trace/exe/Program.cs

```diff
@@ -229,11 +232,41 @@ public void UpdateToState(RequestState state, ulong time, bool allowPrevious = f
                 State = state;
             }
 
+            public void FinalizeState(ulong time)
+            {
+                if (EncounteredError) return;
+                if (time < LastStateChangeTime)
+                {
+                    Console.WriteLine("ERROR: Invalid final time while in {0}", State);
+                    EncounteredError = true;
+                    return;
+                }
+                Times[(int)State] += (time - LastStateChangeTime);
+                LastStateChangeTime = time;
+            }
+
             //
             // Returns the sum of all the calculated times for individual layers.
             //
             public ulong TotalTime { get { return Times.Aggregate((temp, x) => temp + x); } }
 
+            //
+            // An estimate of the network cost from (Client.FirstSend-Client.FirstRecv)-(Server.FirstRecv-Server.FirstSend)
```

Comment:
(Client.FirstSend-Client.FirstRecv) => (Client.FirstRecv - Client.FirstSend)?

Reply:
Nope. It's correct as written.

Reply:
Oh, actually both are reversed.